### PR TITLE
feat(라프텔): new presence

### PR DESCRIPTION
**Pull request by remuring9999**
## Description 
As one of the OTT platforms in service in Korea, it is an animation streaming service.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

스크린샷 2023-07-25 195908
스크린샷 2023-07-25 195800
스크린샷 2023-07-25 195855

</details>

### CODE REVIEW

**Comment by DarkVillager** [comment]
> _ No description provided. _

Please use the format we use.

File: websites/#/라프텔/metadata.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+		"id": "868016688090710067",
+		"name": "remuring_"
+	},
+	"service": "라프텔",
+	"description": {
+		"en": "Korean animation OTT streaming service."
+	},
+	"url": "laftel.net",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/2ew5h9U.jpg",
```
Comment (DarkVillager) [inline websites/#/라프텔/metadata.json:13]: Logo's must be 512x512 as per [our guideliness](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#logo).

File: websites/#/라프텔/presence.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+		`{"${decodeURI(location.search.substring(1))
+			.replaceAll('"', '\\"')
+			.replaceAll("&", '","')
+			.replaceAll("=", '":"')}"}`
+	);
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey:"https://i.imgur.com/2ew5h9U.jpg",
```
Comment (DarkVillager) [inline websites/#/라프텔/presence.ts:40]: Ditto

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/#/라프텔/presence.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+				[presenceData.startTimestamp, presenceData.endTimestamp] =
+					presence.getTimestamps(
+						Math.floor(video.currentTime),
+						Math.floor(video.duration)
+					);
+				presenceData.smallImageKey = Assets.Play;
+				presenceData.smallImageText = "재생중";
+			} else {
+				presenceData.startTimestamp = null;
+				presenceData.endTimestamp = null;
```
Comment (DarkVillager) [inline websites/#/라프텔/presence.ts:165]: We prefer delete over setting to null

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Comment by DarkVillager** [comment]
Also please resolve requested change when they're made.

File: websites/#/라프텔/presence.ts (hunk 3 above)
Comment (remuring9999) [reply websites/#/라프텔/presence.ts]: fix it

File: websites/#/라프텔/presence.ts [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+			.replaceAll("=", '":"')}"}`
+	);
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/PDh4ncE.jpg",
+	};
+
+	if (window.location.pathname === "/") {
```
Comment (theusaf) [inline websites/#/라프텔/presence.ts:43]: Properties of `location` are used many times. Destructure them for simplicity/consistency with other presences. In addition, we prefer `document.location` over `window.location`

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/#/라프텔/metadata.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.9",
+	"author": {
+		"id": "868016688090710067",
+		"name": "remuring_"
+	},
+	"service": "라프텔",
```
Comment (Bas950) [inline websites/#/라프텔/metadata.json:7]:
```suggestion
	"service": "라프텔",
	"altnames": ["laftel"],
```

File: websites/#/라프텔/metadata.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+	"description": {
+		"en": "Korean animation OTT streaming service."
```
Comment (Bas950) [inline websites/#/라프텔/metadata.json:9]: also add a Korean description

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/#/라프텔/metadata.json [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+	"$schema": "https://schemas.premid.app/metadata/1.9",
+	"author": {
+		"id": "868016688090710067",
+		"name": "remuring_"
+	},
+	"service": "라프텔",
+	"altnames": ["laftel"],
+	"description": {
+		"en": "Korean animation OTT streaming service.",
+		"ko": "한국에서 서비스하는 애니 OTT 스트리밍 서비스"
```
Comment (Bas950) [inline websites/#/라프텔/metadata.json:11]:
```suggestion
		"ko_KR": "한국에서 서비스하는 애니 OTT 스트리밍 서비스"
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/#/라프텔/presence.ts [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+					label: `별점 ${AnimeData.meta_info.avg_rating}점`,
+					url: `https://laftel.net/item/${AnimeData.id}/review`,
+				},
+			];
+		}
+	} else if (location.pathname.match(/\/player\/\d*\/\d/)) {
+		const video: HTMLVideoElement = document.querySelector("video");
+		if (video && !isNaN(video.duration)) {
+			if (prevData !== document.location.pathname) {
+				prevData = document.location.pathname;
```
Comment (theusaf) [inline websites/#/라프텔/presence.ts:106]: Properties of `location` and `document.location` are used many times in this code. To simplify it, destructure them.
```ts
const { pathname, search } = document.location;
```

File: websites/#/라프텔/presence.ts [hunk 9]
```diff
@@ -0,0 +1,171 @@
+const presence = new Presence({
+	clientId: "1133347072200949770",
+});
+
+let prevData = "",
+	AnimeData: Detail = {},
+	AnimeDataEpisode: Episode = {};
```
Comment (theusaf) [inline websites/#/라프텔/presence.ts:7]: Variables like these should use `camelCase` by convention.

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/#/라프텔/presence.ts (hunk 9 above)
Comment (remuring9999) [reply websites/#/라프텔/presence.ts]: I did my best

File: websites/#/라프텔/presence.ts [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+	title?: string;
+	subject?: string;
+	episodeNum?: string;
+};
+
+type Obj<T> = {
+	[key: string]: T;
+};
+
+function convertCamel<T>(obj: Obj<unknown>): Obj<T> {
```
Comment (Slowlife01) [inline websites/#/라프텔/presence.ts:33]: I'd say this is rather unnecessary. You should just use eslint ignore on this rule

Comment (theusaf) [reply websites/#/라프텔/presence.ts:33]: The eslint ignore may be unnecessary as it only affects Deepscan, which means that the rule may need to be added like this:
https://github.com/PreMiD/Presences/blob/fcc689d2709c70084758151c7ee00c7822c6ac5c/websites/B/Board%20Game%20Arena/util.ts#L1-L25

We might be able to merge it in and ignore those specific Deepscan issues.

Comment (remuring9999) [reply websites/#/라프텔/presence.ts]: thanks, fix it

**Review by Bas950 — APPROVED** [review]

**Review by theusaf — APPROVED** [review]

**Pull request merged**